apps/dashboard/src/hooks/use-fetch-workflows.ts

```diff
@@ -10,18 +9,15 @@ interface UseWorkflowsParams {
   query?: string;
 }
 
-export function useWorkflows({ limit = 12, offset = 0, query = '' }: UseWorkflowsParams = {}) {
+export function useFetchWorkflows({ limit = 12, offset = 0, query = '' }: UseWorkflowsParams = {}) {
   const { currentEnvironment } = useEnvironment();
 
   const workflowsQuery = useQuery({
     queryKey: [QueryKeys.fetchWorkflows, currentEnvironment?._id, { limit, offset, query }],
-    queryFn: async () => {
-      const { data } = await getV2<{ data: ListWorkflowResponse }>(
-        `/workflows?limit=${limit}&offset=${offset}&query=${query}`
-      );
-      return data;
-    },
+    queryFn: () => getWorkflows({ environment: currentEnvironment!, limit, offset, query }),
     placeholderData: keepPreviousData,
+    enabled: !!currentEnvironment?._id,
+    refetchOnWindowFocus: true,
```

Comment:
This caters to the use case where customers will have their dev and prod environments side by side, and they will sync workflows between environments. They should be able to see the update without a full page reload.